azurerm/internal/services/network/express_route_connection_resource.go

```diff
@@ -0,0 +1,333 @@
+
+	if props := resp.ExpressRouteConnectionProperties; props != nil {
+		d.Set("routing_weight", props.RoutingWeight)
+		d.Set("authorization_key", props.AuthorizationKey)
+		d.Set("enable_internet_security", props.EnableInternetSecurity)
+
+		circuitPeeringID := ""
+		if v := props.ExpressRouteCircuitPeering; v != nil {
+			circuitPeeringID = *v.ID
+		}
+		d.Set("express_route_circuit_peering_id", circuitPeeringID)
+
+		if err := d.Set("routing", flattenExpressRouteConnectionRouting(props.RoutingConfiguration)); err != nil {
+			return fmt.Errorf("setting `routing`: %+v", err)
+		}
+	}
+
+	return nil
+}
+
+func resourceExpressRouteConnectionDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.ExpressRouteConnectionsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ExpressRouteConnectionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	future, err := client.Delete(ctx, id.ResourceGroup, id.ExpressRouteGatewayName, id.Name)
+	if err != nil {
+		return fmt.Errorf("deleting %s: %+v", *id, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for deletion of %s: %+v", *id, err)
+	}
+
+	return nil
+}
+
+func expandExpressRouteConnectionRouting(input []interface{}) *network.RoutingConfiguration {
+	if len(input) == 0 {
+		return &network.RoutingConfiguration{}
+	}
+
+	v := input[0].(map[string]interface{})
+	result := network.RoutingConfiguration{}
+
+	if associatedRouteTableId := v["associated_route_table_id"].(string); associatedRouteTableId != "" {
+		result.AssociatedRouteTable = &network.SubResource{
+			ID: utils.String(associatedRouteTableId),
+		}
+	}
+
+	if propagatedRouteTable := v["propagated_route_table"].([]interface{}); len(propagatedRouteTable) != 0 {
+		result.PropagatedRouteTables = expandExpressRouteConnectionPropagatedRouteTable(propagatedRouteTable)
+	}
+
+	return &result
+}
+
+func expandExpressRouteConnectionPropagatedRouteTable(input []interface{}) *network.PropagatedRouteTable {
+	if len(input) == 0 {
+		return &network.PropagatedRouteTable{}
+	}
+
+	v := input[0].(map[string]interface{})
+
+	result := network.PropagatedRouteTable{}
+
+	if labels := v["labels"].(*schema.Set).List(); len(labels) != 0 {
+		result.Labels = utils.ExpandStringSlice(labels)
+	}
+
+	if routeTableIds := v["route_table_ids"].(*schema.Set).List(); len(routeTableIds) != 0 {
+		result.Ids = expandIDsToSubResources(routeTableIds)
+	}
+
+	return &result
+}
+
+func flattenExpressRouteConnectionRouting(input *network.RoutingConfiguration) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	associatedRouteTableId := ""
+	if input.AssociatedRouteTable != nil && input.AssociatedRouteTable.ID != nil {
+		associatedRouteTableId = *input.AssociatedRouteTable.ID
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"associated_route_table_id": associatedRouteTableId,
```

Comment:
Similar in this ID, could we also format this one using its IDFormatter?

Reply:
Updated